### PR TITLE
Refactor/super fluid

### PR DESCRIPTION
**Pull request by jordanlesich**

### CODE REVIEW

File: src/components/StreamList.jsx [hunk 1]
```diff
@@ -0,0 +1,146 @@
+import React, { useState } from 'react';
+import { useParams } from 'react-router-dom';
+import {
+  Flex,
+  Box,
+  Skeleton,
+  Text,
+} from '@chakra-ui/react';
+
+import { useInjectedProvider } from '../contexts/InjectedProviderContext';
```
Comment (skuhlmann) [inline src/components/StreamList.jsx:10]: not a deal breaker, but we could reorder these imports based on the file structure in the style guide.

File: src/components/SuperTokenListItem.jsx [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+}) => {
+  const handleWithdraw = () => {
+    withdrawSupertoken(tokenAddress);
+  };
+
+  const displayComingSoon = () => {
+    successToast({ title: 'Coming Soon' });
+  };
+  const isLoading = loading?.active && loading?.condition === tokenAddress;
+  const shouldDisableWithdraw = token?.tokenBalance <= 0 || (loading?.active && loading?.condition === tokenAddress);
```
Comment (skuhlmann) [inline src/components/SuperTokenListItem.jsx:35]: another small layout thing. I think we're going with simple statements above functions.

https://daohaus.club/docs/devs/react-layount

File: src/components/superTokenList.jsx [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+
+import { useInjectedProvider } from '../contexts/InjectedProviderContext';
+import { useOverlay } from '../contexts/OverlayContext';
+import { useTX } from '../contexts/TXContext';
+import { useUser } from '../contexts/UserContext';
+
+import ContentBox from './ContentBox';
+import TextBox from './TextBox';
+
+import { createPoll } from '../services/pollService';
```
Comment (skuhlmann) [inline src/components/superTokenList.jsx:17]: check import order

https://daohaus.club/docs/devs/react-layount

File: src/components/superTokenList.jsx [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+            <TextBox size='xs'>Internal Bal.</TextBox>
+          </Box>
+          <Box w='40%' d={['none', null, null, 'inline-block']}>
+            <TextBox size='xs'>Actions</TextBox>
+          </Box>
+        </Flex>
+
+        <Skeleton isLoaded={!loadingStreams}>
+          {superTokenBalances && balanceArr.length > 0
+        && balanceArr?.map(
```
Comment (skuhlmann) [inline src/components/superTokenList.jsx:117]: i see balanceArr.? after balanceArr.length - if balanceArr is undefined, won't that .length call fail?

File: src/utils/general.js [hunk 5]
```diff
@@ -74,6 +75,13 @@ export const detailsToJSON = (values) => {
   if (values.uberType) {
     details.uberType = values.uberType;
   }
+  if (values.ratePerSec) {
```
Comment (skuhlmann) [inline src/utils/general.js:78]: what do you think of this details to JSON checking each key like this?

i've had issues forgetting to add a check when doing a new proposal that needed a different details key.

maybe just something to consider when we do more proposal overhaul.

File: src/components/superTokenList.jsx (hunk 4 above)
Comment (jordanlesich) [reply src/components/superTokenList.jsx:117]: It definitely would.

**Pull request closed without merging**